enhancements/proposals/1978-scylladbdatacenter-api/README.md

```diff
@@ -0,0 +1,761 @@
+
+	// conditions hold conditions describing ScyllaDBDatacenter state.
+	// To determine whether a cluster rollout is finished, look for Available=True,Progressing=False,Degraded=False.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// version is the current version of Scylla in use.
+	// +optional
+	CurrentVersion string `json:"currentVersion,omitempty"`
+
+	// updatedVersion is the updated version of Scylla.
+	// +optional
+	UpdatedVersion string `json:"updatedVersion,omitempty"`
+
+	// nodes is the total number of nodes requested in datacenter.
+	// +optional
+	Nodes *int32 `json:"nodes,omitempty"`
+
+	// nodes is the total number of nodes created in datacenter.
+	CurrentNodes *int32 `json:"currentNodes,omitempty"`
+
+	// updatedNodes is the number of nodes matching the current spec in datacenter.
+	// +optional
+	UpdatedNodes *int32 `json:"updatedNodes,omitempty"`
+
+	// readyNodes is the total number of ready nodes in datacenter.
+	// +optional
+	ReadyNodes *int32 `json:"readyNodes,omitempty"`
+
+	// rackCount is the number of ScyllaDB racks in this cluster.
+	// +optional
+	RackCount *int32 `json:"rackCount,omitempty"`
+
+	// racks reflect the status of datacenter racks.
+	Racks []RackStatus `json:"racks"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:printcolumn:name="AVAILABLE",type=string,JSONPath=".status.conditions[?(@.type=='Available')].status"
+// +kubebuilder:printcolumn:name="PROGRESSING",type=string,JSONPath=".status.conditions[?(@.type=='Progressing')].status"
+// +kubebuilder:printcolumn:name="DEGRADED",type=string,JSONPath=".status.conditions[?(@.type=='Degraded')].status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+
+// ScyllaDBDatacenter defines a monitoring instance for ScyllaDB clusters.
+type ScyllaDBDatacenter struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec defines the desired state of this ScyllaDBDatacenter.
+	Spec ScyllaDBDatacenterSpec `json:"spec,omitempty"`
+
+	// status is the current status of this ScyllaDBDatacenter.
+	Status ScyllaDBDatacenterStatus `json:"status,omitempty"`
+}
+
+```
+
+### Test Plan
+
+All existing E2E's will be changed to use `v1alpha1.ScyllaDBDatacenter`.
+Conversion logic between two APIs is going to be covered by unit tests.
+Additional E2E to validate converting controller.
+
+### Upgrade / Downgrade Strategy
+
+New CRD needs to be installed upon Scylla Operator update. Our documentation and every release notes cover that.
+No specific actions are required for the downgrade.
+
+### Version Skew Strategy
+
+Existing `v1.ScyllaCluster` is going to be converted to `v1alpha.ScyllaDBDatacenter` and controllers are going to work on both of these resources.
```

Comment:
We are not doing "conversion"of resources.

Is this just stale? I thought the plan is to change scyllacluster controller to create ScyllaDBDatacenter for each ScyllaCluster in it's sync loop (with a controllerRef set) and then a new controller (with the modified old sync loops from the old scyllacluster controller) will reconcile ScyllaDBDatacenters, no matter if controller or user created.